File: solutions/go/word-count/1/word_count.go
package wordcount

import (
	"strings"
)

type Frequency map[string]int

func WordCount(phrase string) Frequency {
	test := " " + phrase + " "

	{
		replace := []string{
			",", " '", "' ", ".", ":", "!",
			"#", "%", "&", "(", ")", "*",
			"+", ",", "^", "@", "$", "\n",
		}
		for _, ii := range replace {
			test = strings.ReplaceAll(test, ii, " ")
		}
	}

	test2 := strings.Split(test, " ")

	result := Frequency{}
	for _, ii := range test2 {
		if ii == "" {
			continue
		}
		result[strings.ToLower(ii)]++
	}

	return result
}

File: solutions/lua/word-count/1/word-count.lua
local function word_count(s)
    local result = {}
    s = " " .. s:lower() .. " "
    s = s:gsub("[$!]", " "):gsub("%s'", " "):gsub("'%s", " ")
    for w in s:gmatch("[a-z0-9']+") do result[w] = (result[w] or 0) + 1 end
    return result
end

return {word_count = word_count}
